src/main.rs

```diff
@@ -65,3 +71,35 @@ async fn main() {
 
     info!("application shutdown")
 }
+
+#[allow(dead_code)]
+fn infos() -> Result<()> {
+    info!("bootloader was updated: {}", bootloader_updated());
+    info!("device booted from root {}.", current_root()?);
+    Ok(())
+}
+
+fn current_root() -> Result<&'static str> {
+    let current_root = fs::read_link(DEV_OMNECT.to_owned() + "rootCurrent")
+        .context("current_root: getting current root device")?;
+
+    if current_root
+        == fs::read_link(DEV_OMNECT.to_owned() + "rootA").context("current_root: getting rootA")?
+    {
+        return Ok("a");
+    }
+
+    if current_root
+        == fs::read_link(DEV_OMNECT.to_owned() + "rootB").context("current_root: getting rootB")?
+    {
+        return Ok("b");
+    }
+
+    bail!("current_root: device booted from unknown root")
+}
+
+fn bootloader_updated() -> bool {
+    Path::new(BOOTLOADER_UPDATED_FILE)
+        .try_exists()
+        .is_ok_and(|res| res)
+}
```

Comment:
sorry - better move as pub static method to twin/system_info.rs

Reply:
then if have to change in  twin/mod.rs the "system_info" as pub mod system_info and add "use crate::twin::system_info::SystemInfo;" in the main.rs file to get access to the methods. Not sure if we would like that....

Reply:
methods moved to system_info.rs as discussed; done